### PR TITLE
Update README.md and installation section in readthedocs

### PR DESCRIPTION
**Pull request by Dgadavin**
## Description/Comments
A few sentences describing the overall goals of the pull request's commits / links to relevant github issues.

To consider:

* Is this feature suitable for gcdt or better a gcdt-plugin
* Does this feature have test coverage? (see docs for info about tests)
* Backwards compatibility

## Related PRs
Are there any other PR's on which this depends?

External PR | [title](https://github.com/glomex/gcdt/)

## Checklist before opening PR 
- [ ] Issue / PR description provided
- [ ] This feature has test coverage
- [ ] This feature has documentation (docs)
- [ ] The changelog has been updated (docs/CHANGELOG.md)

## Contributor License Agreement

gcdt is an open-source project that is licensed very permissively (see LICENSE). This means that software, and other contributions to the project are released to the public to be freely used, modified, remixed, and even sold without anyone’s permission. In order to protect this freedom, contributors must agree to waive their rights to their contributions BEFORE contributing them.

By submitting a pull request to gcdt you agree to this Contributor License Agreement, you are waiving your rights to the intellectual property that you contribute to the gcdt project. You agree to put your contribution under MIT license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,65 +40,14 @@ We hope gcdt will be helpful to you, too. At glomex we believe that only open so
...
-Now you can start using gcdt:
-
-``` bash
-$ gcdt version
-```
-
-BTW, `gcdt version` shows you all the versions of gcdt and installed plugins. So you can use this to quickly check which plugins are installed. 
+## Installation
 
+Follow the [instruction](http://gcdt.readthedocs.io/en/latest/07_installation.html) to install `gcdt`
```
Comment (jbrandstetter) [inline README.md]: should be "instructions"

File: docs/07_installation.md [hunk 2]
```diff
@@ -37,56 +37,55 @@ It is a very common practice not to install Python packages by hand. Instead dep
 
 The grouping is not enforced by packaging but to have a std. within an organization is beneficial especially if your want to reuse CI/CD tools.
 
-A little opinionated but pretty common:
-
-* `requirements.txt` tools and packages your service directly depends on
-* `requirements_def.txt` tools and packages you need to develop and test your service
```
Comment (jbrandstetter) [inline docs/07_installation.md]: should be requirements_dev.txt

File: docs/07_installation.md [hunk 3]
```diff
@@ -37,56 +37,55 @@ It is a very common practice not to install Python packages by hand. Instead dep
...
-* `requirements_gcdt.txt` gcdt and gcdt plugins you use to deploy your service to AWS (see https://github.com/glomex/gcdt#installing-gcdt)
-* `requirements_docs.txt` tools you need to write and build your documentation
-
-TODO: document version schema
-
-* add gcdt to your requirements_gcdt.txt
-* add the plugins you use to requirements_gcdt.txt
-
```
Comment (jbrandstetter) [inline docs/07_installation.md]: i would keep the information in that section as it's crucial for how we install gcdt

File: docs/07_installation.md [hunk 4]
```diff
@@ -37,56 +37,55 @@ It is a very common practice not to install Python packages by hand. Instead dep
...
+gcdt-gen-serverless
+```
 
+This is also a best practice to use the `requirements_gcdt.txt` file on your build server.
 
-TODO add private repo deps
+### Prepare virtualenv
 
-All gcdt packages live in a private PyPi repository. See [reposerver](http://reposerver-prod-eu-west-1.infra.glomex.cloud/pypi/) for instructions.
+I am sure every Python dev uses virtualenv on a day to day basis. But we also use gcdt to deploy PHP, Ruby, and NodeJs projects. So I like to cover the basics:
```
Comment (jbrandstetter) [inline docs/07_installation.md]: Should say "we" not "I". I'd rephrase to "Using virtualenvs for Python is considered best practise. This is what you need to do:
And Mark's previous explanation is pretty good, so let's keep it:
create a virtualenv ('$ virtualenv venv')
install the packages you want to use (see below)
a virtualenv works basically like every other technical device, you need to switch it on before you can use it ('$ source ./venv/bin/activate')

File: docs/07_installation.md (hunk 2 above)
Comment (Dgadavin) [inline docs/07_installation.md]: We delete this part.

File: docs/60_faq.md [hunk 5]
```diff
@@ -11,25 +11,67 @@ must supply either home or prefix/exec-prefix -- not both
 
 You can find a solution on [here](http://stackoverflow.com/questions/24257803/distutilsoptionerror-must-supply-either-home-or-prefix-exec-prefix-not-both)
 
+### Raumel package error
```
Comment (jbrandstetter) [inline docs/60_faq.md]: make this a more generic error and not specific to ruamel. happens to more packages than only ruamel

File: docs/60_faq.md [hunk 6]
```diff
@@ -11,25 +11,67 @@ must supply either home or prefix/exec-prefix -- not both
...
+```
+you need to upgrade `pip` with latest version:
+```bash
+$ pip install pip --upgrade
+```
+
+### Bundling error
 
-Be sure to provide the correct environment variables (ENV=PROD/DEV/etc.)
+Such error usually appear because you didn't install `gcdt-bundler` plugin:
```
Comment (jbrandstetter) [inline docs/60_faq.md]: This error is usually caused by not having installed the gcdt-bundler plugin.

File: docs/07_installation.md (hunk 2 above)
Comment (jbrandstetter) [inline docs/07_installation.md]: I'd keep it. Makes sense to describe the motivation behind requirements_gcdt.txt

File: docs/07_installation.md [hunk 7]
```diff
@@ -1,92 +1,91 @@
...
 
-### Installation
-
-#### Setup virtualenv
+This is also a best practice to use the `requirements_gcdt.txt` file on your build server.
 
-Using virtualenv is a little bit evolved so some people do not like to use it. There is three things you need to do.
+### Setup virtualenv
 
+We sure every Python dev uses virtualenv on a day to day basis. Using virtualenvs for Python is considered best practise. This is what you need to do:
```
Comment (jbrandstetter) [inline docs/07_installation.md]: delete the first sentence, doesn't add any valuable information

File: docs/60_faq.md [hunk 8]
```diff
@@ -11,25 +11,88 @@ must supply either home or prefix/exec-prefix -- not both
 
 You can find a solution on [here](http://stackoverflow.com/questions/24257803/distutilsoptionerror-must-supply-either-home-or-prefix-exec-prefix-not-both)
 
+### Python package errors
 
-### Environment variables
+**Please ensure that you have the latest version of `pip` and `virtualenv`**
```
Comment (jbrandstetter) [inline docs/60_faq.md]: and setuptools

File: docs/60_faq.md [hunk 9]
```diff
@@ -11,25 +11,88 @@ must supply either home or prefix/exec-prefix -- not both
...
+  File "/var/lib/jenkins/workspace/Infrastructure/mes-ftp/gcdtauto/venv/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 918, in require
+    needed = self.resolve(parse_requirements(requirements))
+  File "/var/lib/jenkins/workspace/Infrastructure/mes-ftp/gcdtauto/venv/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 805, in resolve
+    raise DistributionNotFound(req)
+pkg_resources.DistributionNotFound: regex==2017.6.07
+```
+Update `virtualenv`:
+```bash
+$ pip install virtualenv --upgrade
+```
```
Comment (jbrandstetter) [inline docs/60_faq.md]: the basic error "pkg_resources.DistributionNotFound" is the same, so you can make this section shorter by just combining:
"In case of errors like these:"
....
"Usually updating pip and virtualenv solves the problem:"

And then be consistent about the pip call - sometimes it's --upgrade and sometimes -U  We should stick to one syntax

File: docs/07_installation.md [hunk 10]
```diff
@@ -41,52 +43,60 @@ A little opinionated but pretty common:
 
 * `requirements.txt` tools and packages your service directly depends on
 * `requirements_def.txt` tools and packages you need to develop and test your service
-* `requirements_gcdt.txt` gcdt and gcdt plugins you use to deploy your service to AWS (see https://github.com/glomex/gcdt#installing-gcdt)
-* `requirements_docs.txt` tools you need to write and build your documentation
```
Comment (jbrandstetter) [inline docs/07_installation.md]: what about the docs stuff?

File: docs/40_ramuda.md [hunk 11]
```diff
@@ -43,28 +41,24 @@ Options:
...
 
-
 #### bundle
 zips all the files belonging to your lambda according to your config and requirements.txt and puts it in your current working directory as `bundle.zip`. Useful for debugging as you can still provide different environments.
 
-
 #### deploy
 
-Deploy a AWS Lambda function to AWS. If the lambda function is non-existent it will create a new one. 
+Deploy a AWS Lambda function to AWS. If the lambda function is non-existent it will create a new one.
```
Comment (jbrandstetter) [inline docs/40_ramuda.md]: ...an AWS Lambda...

File: docs/60_faq.md [hunk 12]
```diff
@@ -11,25 +11,74 @@ must supply either home or prefix/exec-prefix -- not both
...
+pip._vendor.pkg_resources.DistributionNotFound:
+```
+or
+```bash
+pkg_resources.DistributionNotFound: regex==2017.6.07
+```
+you should update your `pip` and `virtualenv` packages
+```bash
+$ pip install pip --upgrade
+$ pip install virtualenv --upgrade
```
Comment (jbrandstetter) [inline docs/60_faq.md]: still not consistent, --upgrade or -U

File: docs/60_faq.md [hunk 13]
```diff
@@ -11,25 +11,74 @@ must supply either home or prefix/exec-prefix -- not both
...
+KeyError: u'_zipfile'
+```
+You need to add `gcdt-bundler` into *requirements_gcdt.txt* and do:
+```bash
+$ pip install -U -r requirements_gcdt.txt
+```
+
+### Missing configuration error
+
+After updating `gcdt` to the latest version you saw such error:
```
Comment (jbrandstetter) [inline docs/60_faq.md]: After updating `gcdt` to the latest version you get the following error:

File: docs/60_faq.md [hunk 14]
```diff
@@ -11,25 +11,74 @@ must supply either home or prefix/exec-prefix -- not both
...
+```bash
+Configuration missing for ‘kumo’
+```
+This error appear if you used `hocon` based configs without installed `glomex-config-reader` plugin. You can install it or use [conf2json](http://sre-docs.glomex.cloud/glomex-config-reader/userguide/40_glomex_config_reader.html#command-conf2json) util (only for glomex users) to transform your `hocon` configs into `json` one.
```
Comment (jbrandstetter) [inline docs/60_faq.md]: ...appears... without having installed the ...

File: docs/60_faq.md [hunk 15]
```diff
@@ -11,25 +11,74 @@ must supply either home or prefix/exec-prefix -- not both
...
 
+### Environment variable error
+
+If you run any `gcdt` commands (kumo, tenkai, ramuda etc) and have such error:
```
Comment (jbrandstetter) [inline docs/60_faq.md]: ... get the following error:

File: docs/07_installation.md (hunk 10 above)
Comment (Dgadavin) [inline docs/07_installation.md]: We have this in Development section https://www.youtube.com/watch?v=8zusa6drjH8 Do we need it in Installation section?
